lib/resty/iprepd.lua

```diff
@@ -0,0 +1,76 @@
+local cjson = require('cjson')
+local lrucache = require('resty.lrucache')
+local http = require('resty.http')
+
+function fatal_error(error_msg)
+  ngx.log(ngx.ERR, error_msg)
+  os.exit(1)
+end
+
+local _M = {}
+local mt = { __index = _M }
+
+function _M.new(options)
+  local iprepd_url = options.url or 'http://localhost:8080/'
+  local cache_ttl = options.cache_ttl or 30
+  local iprepd_threshold = options.threshold
+  if not iprepd_threshold then
+    fatal_error('Need to pass in a threshold')
```

Comment:
In `ngx.log()` does it note which module created the failure? Or should we include more information stating it's from iprepd.lua in the error message?

Reply:
Yes, it includes what file it came from:

```
2018/10/22 16:49:23 [error] 1#1: [lua] iprepd.lua:6: fatal_error(): Need to pass in an api_key                                                                                                        
```